### PR TITLE
Multiple value selection for enumeration : Fix #282

### LINKED ISSUE #282: [ParameterEditor] Multiple value for Enumeration
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

The component _EnumerationParameterTypeEditor_ should allow the multi-selection when enabled

- [x] Make sure the UI facilitates multi select (use ListBox component)
- [x] Read and save values from UI to database / COMET-SDK

### Logs

  - [ ] There are no logs regarding the issue
  - { ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- COMET version:
  - [ ] COMET-WEB: 
  - [ ] Other: 
- Web Services:
  - [ ] COMET Web Services version:

### PR DESCRIPTION
**Pull request by nabilatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fix #282 
For EnumerationParameterType that allows multi-selection :
- The UI shows a dropdown component with a ListBox to select multiple elements
- Write selected values to database

### CODE REVIEW

**Review by jaimeatrhea — APPROVED** [review]

File: COMETwebapp.Tests/ViewModels/Components/Shared/ParameterEditors/EnumerationParameterTypeEditorViewModelTestFixture.cs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        public void VerifyConfirmSelection()
+        {
+            this.viewModel.OnSelectAllChanged(true);
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(this.viewModel.SelectedEnumerationValueDefinitions, Is.Not.Null);
+                Assert.That(this.viewModel.SelectedEnumerationValueDefinitions.ToList(), Has.Count.EqualTo(3));
+                Assert.That(this.viewModel.IsOnEditMode, Is.False);
+                Assert.That(async () => this.viewModel.OnConfirmButtonClick(), Throws.Nothing);
```
Comment (antoineatrhea) [inline COMETwebapp.Tests/ViewModels/Components/Shared/ParameterEditors/EnumerationParameterTypeEditorViewModelTestFixture.cs:159]: please await the method

File: COMETwebapp/Components/Shared/ParameterTypeEditors/EnumerationParameterTypeEditor.razor.cs [hunk 2]
```diff
@@ -52,6 +52,15 @@ protected override void OnParametersSet()
         {
             base.OnParametersSet();
 
+            if (this.ViewModel is IEnumerationParameterTypeEditorViewModel vm)
+            {
+                this.Disposables.Add(vm.WhenAnyValue(x => x.IsOnEditMode)
```
Comment (antoineatrhea) [inline COMETwebapp/Components/Shared/ParameterTypeEditors/EnumerationParameterTypeEditor.razor.cs:57]: please combine those registration

File: COMETwebapp/ViewModels/Components/Shared/ParameterEditors/IEnumerationParameterTypeEditorViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//  </copyright>
+//  --------------------------------------------------------------------------------------------------------------------
+
+namespace COMETwebapp.ViewModels.Components.Shared.ParameterEditors
+{
+    using CDP4Common.SiteDirectoryData;
+
+    public interface IEnumerationParameterTypeEditorViewModel
```
Comment (antoineatrhea) [inline COMETwebapp/ViewModels/Components/Shared/ParameterEditors/IEnumerationParameterTypeEditorViewModel.cs:29]: Missing documentation

**Review by antoineatrhea — CHANGES_REQUESTED** [review]
Small changes requested

**Review by antoineatrhea — APPROVED** [review]

**Pull request merged**